pkg/services/sso/redhatsso_service.go

```diff
@@ -127,9 +202,9 @@ func (r *redhatssoService) DeRegisterKasFleetshardOperatorServiceAccount(accessT
 		}
 	}
 
-	err := r.client.DeleteServiceAccount(accessToken, agentClusterId)
+	err := r.DeleteServiceAccount(accessToken, context.Background(), agentClusterId)
```

Comment:
why this change?

Reply:
Because this way I reuse the code in `DeleteServiceAccount` to retrieve the error message from the error returned by `redhat_sso`.

Reply:
make sense